core/src/com/unciv/ui/worldscreen/bottombar/BattleTable.kt

```diff
@@ -167,39 +179,104 @@ class BattleTable(val worldScreen: WorldScreen): Table() {
+        val attackText : String = when (attacker) {
+            is CityCombatant -> "Bombard"
             else -> "Attack"
         }
         val attackButton = TextButton(attackText.tr(), skin).apply { color= Color.RED }
 
-        var attackableEnemy : AttackableTile? = null
+        var attackableTile : AttackableTile? = null
 
         if (attacker.canAttack()) {
             if (attacker is MapUnitCombatant) {
-                attackableEnemy = battleHelper
+                attackableTile = battleHelper
                         .getAttackableEnemies(attacker.unit, attacker.unit.movement.getDistanceToTiles())
                         .firstOrNull{ it.tileToAttack == defender.getTile()}
             }
             else if (attacker is CityCombatant)
             {
                 val canBombard = UnitAutomation().getBombardTargets(attacker.city).contains(defender.getTile())
                 if (canBombard) {
-                    attackableEnemy = AttackableTile(attacker.getTile(), defender.getTile())
+                    attackableTile = AttackableTile(attacker.getTile(), defender.getTile())
                 }
             }
         }
 
-        if (!worldScreen.isPlayersTurn || attackableEnemy == null) {
+        if (!worldScreen.isPlayersTurn || attackableTile == null) {
             attackButton.disable()
             attackButton.label.color = Color.GRAY
         }
 
         else {
             attackButton.onClick {
                 try {
-                    Battle.moveAndAttack(attacker, attackableEnemy)
+                    Battle.moveAndAttack(attacker, attackableTile)
+                    worldScreen.mapHolder.unitActionOverlay?.remove() // the overlay was one of attacking
+                    worldScreen.shouldUpdate = true
+                }
+                catch (ex:Exception){
+                    val battleBugPopup = Popup(worldScreen)
+                    battleBugPopup.addGoodSizedLabel("You've encountered a bug that I've been looking for for a while!").row()
+                    battleBugPopup.addGoodSizedLabel("If you could copy your game data (\"Copy saved game to clipboard\" - ").row()
+                    battleBugPopup.addGoodSizedLabel("  paste into an email to yairm210@hotmail.com)").row()
+                    battleBugPopup.addGoodSizedLabel("It would help me figure out what went wrong, since this isn't supposed to happen!").row()
+                    battleBugPopup.addGoodSizedLabel("If you could tell me which unit was selected and which unit you tried to attack,").row()
+                    battleBugPopup.addGoodSizedLabel("  that would be even better!").row()
+                    battleBugPopup.open()
+                }
+            }
+        }
+
+        add(attackButton).colspan(2)
+
+        pack()
+
+        setPosition(worldScreen.stage.width/2-width/2, 5f)
+    }
+
+    private fun simulateNuke(attacker: MapUnitCombatant, targetTile: TileInfo){
+        clear()
+        defaults().pad(5f)
```

Comment:
I see that we now have defaults().pad(5f) twice and row().pad(5f) also twice, this should just be a single defaults().pad(5f) in the init function